pkg/reconciler/pipelinerun/resources/pipelineref.go

```diff
@@ -42,36 +42,42 @@ func GetPipelineFunc(ctx context.Context, k8s kubernetes.Interface, tekton clien
 	cfg := config.FromContextOrDefaults(ctx)
 	pr := pipelineRun.Spec.PipelineRef
 	namespace := pipelineRun.Namespace
-	// if the spec is already in the status, do not try to fetch it again, just use it as source of truth
+	// if the spec is already in the status, do not try to fetch it again, just use it as source of truth.
+	// Same for the Source field in the Status.Provenance.
 	if pipelineRun.Status.PipelineSpec != nil {
-		return func(_ context.Context, name string) (v1beta1.PipelineObject, error) {
+		return func(_ context.Context, name string) (v1beta1.PipelineObject, *v1beta1.ConfigSource, error) {
+			var configSource *v1beta1.ConfigSource
+			if pipelineRun.Status.Provenance != nil {
+				configSource = pipelineRun.Status.Provenance.ConfigSource
```

Comment:
This is not tested. Can we test it?

Reply:
I think it's covered in [TestGetPipelineFuncSpecAlreadyFetched](https://github.com/tektoncd/pipeline/pull/5397/files/074210dd2eee8df89d5a15d9ceb42e3a89fec178#diff-3864741a4c79caf7dcb0d45d46fd2780275657d519554f0fd14dd56adc9f1fdaL209-R282) same for taskrun level in [TestGetTaskFuncFromTaskRunSpecAlreadyFetched](https://github.com/tektoncd/pipeline/pull/5397/files/074210dd2eee8df89d5a15d9ceb42e3a89fec178#diff-bd1e986450e0c75da7f92a1a33b2bff6a133d63c3d584dedb90d1e8abce509d5L451-R533). 

PTAL and let me know if this is not what you meant. Thank you!!

Reply:
The coverage report says it is not tested. I didn't look into the details maybe you're right